src/ophyd_async/core/_device.py

```diff
@@ -71,13 +71,16 @@ class Device(HasName, Connectable):
     _connect_task: asyncio.Task | None = None
     # The mock if we have connected in mock mode
     _mock: LazyMock | None = None
+    # The separator to use when making child names
+    _child_name_separator: str = "-"
 
     def __init__(
         self, name: str = "", connector: DeviceConnector | None = None
     ) -> None:
         self._connector = connector or DeviceConnector()
         self._connector.create_children_from_annotations(self)
-        self.set_name(name)
+        if name:
+            self.set_name(name)
```

Comment:
Nit: Without the refactor this change is a bit redundant? Unless we don't trust the the type system? But it does no harm

Reply:
It is a tiny performance improvement to not set the child names if the parent name isn't set